src/Cognito/Provider.php

```diff
@@ -0,0 +1,170 @@
+    }
+
+    /**
+     * Get the token URL for the provider.
+     *
+     * @return string
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getCognitoUrl('token_uri');
+    }
+
+    /**
+     * Get the raw user for the given access token.
+     *
+     * @param string $token
+     *
+     * @return array
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getCognitoUrl('userinfo_uri'), [
+            'headers' => [
+                'Authorization' => 'Bearer '.$token,
+            ],
+        ]);
+
+        return (array) json_decode($response->getBody(), true);
+    }
+
+    /**
+     * Map the raw user array to a Socialite User instance.
+     *
+     * @param array $user
+     *
+     * @return \Laravel\Socialite\User
+     */
+    protected function mapUserToObject(array $user)
+    {
+        $key = $this->getConfig('userinfo_key', null);
+        $data = is_null($key) === true ? $user : Arr::get($user, $key, []);
+
+        return (new User())->setRaw($data)->map([
+            'id'       => $this->getUserData($data, 'sub'),
+            'nickname' => $this->getUserData($data, 'nickname'),
+            'name'     => $this->getUserData($data, 'name'),
+            'email'    => $this->getUserData($data, 'email'),
+            'avatar'   => $this->getUserData($data, 'picture'),
+
+            // user - array containing all available user attributes
+            // Cognito adheres to OIDC standard claims - https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
+        ]);
+    }
+
+    /**
+     * Get the POST fields for the token request.
+     *
+     * @param string $code
+     *
+     * @return array
+     */
+    protected function getTokenFields($code)
+    {
+        return array_merge(parent::getTokenFields($code), [
+            'grant_type' => 'authorization_code',
+        ]);
+    }
+
+    protected function getCognitoUrl($type)
+    {
+        return rtrim($this->getConfig('host'), '/').'/'.ltrim(($this->getConfig($type, Arr::get([
+            'authorize_uri' => 'oauth2/authorize',
+            'token_uri'     => 'oauth2/token',
+            'userinfo_uri'  => 'oauth2/userInfo',
+        ], $type))), '/');
+    }
+
+    protected function getUserData($user, $key)
+    {
+        return Arr::get($user, $this->getConfig('user_'.$key, $key));
+    }
+
+    // Logout the user from cognito (expire SSO tokens), then redirect to url.
+    public function logoutCognitoUser()
+    {
+        $authHost = config('services.cognito.host');
+        $clientId = config('services.cognito.client_id');
+        $logoutUri = config('services.cognito.logout_uri');
+
+        return "$authHost/logout?client_id=$clientId&logout_uri=$logoutUri";
+    }
+
+    // Logout the user from cognito (expire SSO tokens), then redirect the user to he sign in page.
+    public function switchCognitoUser()
```

Comment:
missing `@return` docblock

Reply:
I think this is now done, I referenced another provider.